.circleci/config.yml

```diff
@@ -0,0 +1,315 @@
+            python3 -m pip install *.tar.gz
+            python3 -c "$PYPI_SMOKE_CODE"
+            deactivate
+
+  deploy_pypi_wheel_build:
+    description: "Build wheel packages for pypi"
+    steps:
+      - run:
+          name: create packages
+          command: |
+            . venv/bin/activate
+            export DIST_EXTRA_CONFIG=/tmp/build-opts.cfg
+            echo -e "[bdist_wheel]\npython_tag=$PYTHON_TAG" > $DIST_EXTRA_CONFIG
+            python -m build --wheel
+
+      - run:
+          name: smoke packages
+          command: |
+            mkdir -p /tmp/wheeltest
+            python3 -m venv /tmp/wheeltest/venv
+            cp dist/*.whl /tmp/wheeltest
+            cd /tmp/wheeltest
+            . ./venv/bin/activate
+            python3 -m pip install -U wheel pip twine setuptools
+            python3 -m twine check *.whl
+            python3 -m pip install *.whl
+            python3 -c "$PYPI_SMOKE_CODE"
+            deactivate
+
+  deploy_pypi_upload:
+    description: "Upload packages to pypi"
+    steps:
+      - run:
+          name: upload to pypi
+          command: |
+            . venv/bin/activate
+            twine upload dist/*
+
+jobs:
+
+  python3:
+    parameters:
+      python-version:
+        type: string
+
+    resource_class: xlarge
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+        environment:
+          PYVERS: << parameters.python-version >>
+          COVERAGE_FILE: test-reports/<< pipeline.git.revision >>/.coverage
+          COVERAGE_ARGS: --cov yyjson --cov-append
+
+    working_directory: ~/repo
+
+    steps:
+      - test_steps_python
+
+  deploy_pypi:
+    parameters:
+      python-version:
+        type: string
+      python-tag:
+        type: string
+
+    docker:
+      - image: << parameters.python-version >>
+        environment:
+          PYPI_SMOKE_CODE: import yyjson; print(yyjson.__package__)
+          PYTHON_TAG: << parameters.python-tag >>
+
+    steps:
+      - deploy_pypi_prologue
+      - deploy_pypi_tag_verify
+      - deploy_pypi_wheel_build
+      - deploy_pypi_sdist_build
+      - deploy_pypi_upload
+
+workflows:
+  version: 2.1
+  run_tests:
+    unless:
+      equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
+    jobs:
+      - python3:
+          name: python311
+          python-version: "3.11"
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: /.*/
+
+      - python3:
+          name: python313
+          python-version: "3.13"
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: /.*/
+
+      - deploy_pypi:
+          name: deploy311
+          python-tag: py311
+          python-version: "3.11"
+          requires:
+            - python311
+            - python313
+          context:
+            - PublicPypiAccess
+          filters:
+            tags:
+              only: /^v2\.[0-9]+\.[0-9]+((a|b|rc)[0-9]*)?$/
```

Comment:
Are these `tag` filters the correct values we'll want to use?

Reply:
fixed.